ReactAndroid/src/main/java/com/facebook/react/modules/image/ImageLoaderModule.java

```diff
@@ -45,6 +48,48 @@ public String getName() {
     return "ImageLoader";
   }
 
+  @ReactMethod
+  public void getSize(
+      String uriString,
+      final Callback successCallback,
```

Comment:
Make the public JS APIs use callbacks, but use promises for the private native implementation.